seisbench/generate/labeling.py

```diff
@@ -774,6 +776,208 @@ def __str__(self):
+                Otherwise, a 2-tuple is expected, with the first string indicating the key to
+                read from and the second one the key to write to.
+                Default key is ("X", "y").
+    :param nfft: Length of the FFT used, if a zero padded FFT is desired for scipy STFT.
+                 If None, the FFT length is nperseg.
+    :param nperseg: Length of each segment for scipy STFT. Default is 60
+    :param eps: Avoiding division by zero, default is 1e-12.
+    """
+
+    def __init__(
+        self,
+        noise_dataset: seisbench.data.base.WaveformDataset,
+        scale: tuple[float, float] = (0, 1),
+        scaling_type: str = "peak",
+        component: str = "ZNE",
+        sampling_rate: Union[float, None] = None,
+        nfft: int = 60,
+        nperseg: int = 30,
+        eps: float = 1e-12,
+        key: tuple[str, str] = ("X", "y"),
+        **kwargs,
+    ):
+        super().__init__(label_type="binary", dim=2)
+
+        self.noise_dataset = noise_dataset
+        self.scale = scale
+        self.scaling_type = scaling_type.lower()
+        self.key = key
+        self.component = component
+        self.sampling_rate = sampling_rate
+        self.noise_generator = seisbench.generate.GenericGenerator(noise_dataset)
+        self.noise_samples = len(noise_dataset)
+        self.nfft = nfft
+        self.nperseg = nperseg
+        self.eps = eps
+        self.kwargs = kwargs
+
+        if self.scaling_type.lower() not in ["std", "peak"]:
+            msg = "Argument scaling_type must be either 'std' or 'peak'."
+            raise ValueError(msg)
+
+    def __call__(self, state_dict):
+        x, metadata = state_dict[self.key[0]]
+
+        if self.key[0] != self.key[1]:
+            # Ensure data and metadata is not modified inplace unless input and output key are anyhow identical
+            metadata = copy.deepcopy(metadata)
+            x = x.copy()
+
+        # Select component if only one is given. If more than one component is given, choose random component
+        # Note, if more than one component is given, components of signal and noise are not mixed
+        if len(self.component) == 1:
+            component_idx = metadata["trace_component_order"].index(self.component)
+            metadata["trace_component_order"] = self.component
+        else:
+            component_idx = random.randint(0, len(self.component) - 1)
+            metadata["trace_component_order"] = self.component[component_idx]
+
+        # Select component from x and modify metadata
+        x = x[component_idx, :]
+
+        # Defining scale for earthquake and noise amplitude
+        # I.e., scale for earthquake and noise can become, for example, zero and thus either no noise or
+        # earthquake waveform exists. Using this approach, the denoiser learns better to distinguish between noise
+        # and earthquake.
+        scale_earthquake = np.random.uniform(*self.scale)
```

Comment:
I'm not sure this is consistent with the docstring. From the docstring I assumed that the scale describes the relative amplitude of the noise compared to the signal. However, here both are scaled. Maybe this should be described explicitly in the docstring?